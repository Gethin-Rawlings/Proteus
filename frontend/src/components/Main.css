.main {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows:  60px 1fr 40px;
    grid-template-areas: 
                          "nav  "
                          "main  "
                          "footer  ";
    grid-row-gap: 50px;
    min-height: 100vh;
    background-color: #F1F8FD;
  }
  .main-nav{
    display: flex; 
    max-height: 100% ;
    padding-top: 12px;
    padding-bottom: 8px;
    background-color: #33619D;
    grid-area: nav;
    color:white;
  }
  ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
}
li { 
    margin: 2px;
    display: block;  
}
.logout{
    margin-left: auto;
}
a {
    color:white;
    text-decoration: none;
    padding: 23px;
}
a:hover{
    color: grey;
}
.main-body{
    grid-area: main;
}
.main-footer{
    display: flex;
    padding-top: 12px;
    padding-left: 10px;
    background-color: #33619D;
    color: white;
    grid-area: footer;
    font-size: .8em;
    text-align: left;
  }
  .App-intro{
    grid-area: main;
  }
  .useradmin{
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;    
  }
  .programmeSearch{
    display: flex;
    flex-wrap: wrap;  
  }
.networks{
    text-align: left;
    margin-left: 10px; 
    margin-right: 10px; 
    max-width: 240px;
    }
.indies{
    text-align: left;
    margin-left: 10px; 
    margin-right: 10px; 
    max-width: 240px;
}
.productions{
    text-align: left;
    margin-left: 10px; 
    margin-right: 10px; 
    max-width: 240px;
}
.from{
  text-align: center;
  margin-left: 10px; 
  margin-right: 10px; 
  max-width: 240px;
}
.to{
  text-align: center;
  margin-left: 10px; 
  margin-right: 10px; 
  max-width: 240px;
}
.usersearch{
    text-align: center;
    margin-left: 10px; 
    margin-right: 10px;  
}
option{
    font-family: arial, sans-serif;
}
.dropdown-content {
  border-radius: 5px;
  display: none;
  position: absolute;
  background-color: #f9f9f9;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
  font-size: .8em;
}
.dropdown-content a {
  color: black;
  padding:5px;
  text-decoration: none;
  display: block;
  text-align: left;
  border-radius: 5px;
}
.dropdown-content a:hover {background-color: #f1f1f1}

.dropdown:hover .dropdown-content {
  display: block;
}