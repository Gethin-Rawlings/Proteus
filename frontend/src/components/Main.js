import React from 'react';
import "./Main.css";
import Navbar from "./Navbar"
import Footer from './Footer'

class Main extends React.Component {
  
  render() {
    return (
      <div className="main">
        <Navbar/>
        <section className="main-body">Grumpy cat was ere again
        </section>
        <Footer/>
      </div>

    );
  }
}
export default Main;
