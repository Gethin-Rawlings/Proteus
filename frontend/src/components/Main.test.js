
import React from 'react';
import * as Enzyme from 'enzyme';
import { shallow } from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
import Main from './Main'

describe('storage', () => [localStorage, sessionStorage].map(storage => {
    beforeEach(() => {
      storage.clear();
      jest.clearAllMocks();
    });
  
    Enzyme.configure({ adapter: new ReactSixteenAdapter() });
  
    const history = ['/Main'];
  
    test('Main component should render as expected', () => {
      const KEY = 'loggedIn',
        VALUE = true;
      sessionStorage.setItem(KEY, VALUE);
      const component = shallow(<Main history={history} />);
      const tree = component;
      expect(tree).toMatchSnapshot();
    })
  }));
  
  
  
  
//const history = ['/Main'];
//const spy = jest.spyOn(Main.prototype, 'componentDidMount');
//const wrapper = mount(<BrowserRouter><Main history={history} /></BrowserRouter>);
//console.log( wrapper.debug() )
//wrapper.instance().methodName();
//expect(spy).toHaveBeenCalled();