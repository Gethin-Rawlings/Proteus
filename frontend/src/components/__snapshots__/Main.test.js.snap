// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storage Main component should render as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    history={
        Array [
            "/Main",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Navbar) />,
        <section
          className="main-body"
>
          Grumpy cat was ere again
</section>,
        <Footer />,
      ],
      "className": "main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Grumpy cat was ere again",
          "className": "main-body",
        },
        "ref": null,
        "rendered": "Grumpy cat was ere again",
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Navbar) />,
          <section
            className="main-body"
>
            Grumpy cat was ere again
</section>,
          <Footer />,
        ],
        "className": "main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Grumpy cat was ere again",
            "className": "main-body",
          },
          "ref": null,
          "rendered": "Grumpy cat was ere again",
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`storage Main component should render as expected 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    history={
        Array [
            "/Main",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Navbar) />,
        <section
          className="main-body"
>
          Grumpy cat was ere again
</section>,
        <Footer />,
      ],
      "className": "main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Grumpy cat was ere again",
          "className": "main-body",
        },
        "ref": null,
        "rendered": "Grumpy cat was ere again",
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Navbar) />,
          <section
            className="main-body"
>
            Grumpy cat was ere again
</section>,
          <Footer />,
        ],
        "className": "main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Grumpy cat was ere again",
            "className": "main-body",
          },
          "ref": null,
          "rendered": "Grumpy cat was ere again",
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
